### PR TITLE
[YUNIKORN-593]Optimize the UT for ListApplications

### PR DESCRIPTION
**Pull request by HuangTing-Yao**
### What is this PR for?
Optimize the UT for ListApplications.

### What type of PR is it?
* [ ] - Bug Fix
* [x] - Improvement
* [ ] - Feature
* [ ] - Documentation
* [ ] - Hot Fix
* [ ] - Refactoring

### Todos
* [ ] - Task

### What is the Jira issue?
https://issues.apache.org/jira/browse/YUNIKORN-593

### Questions:
* [ ] - The licenses files need update.
* [ ] - There is breaking changes for older versions.
* [ ] - It needs documentation.

### CODE REVIEW

**Comment by HuangTing-Yao** [comment]
I can move `podCase` array out of Test function, and turn on the lint check, if it is necessary.

**Review by craigcondit — CHANGES_REQUESTED** [review]
Looks good, with the exception of one typo I found.

**Review by wilfred-s — APPROVED** [review]
approved when the typo for the Templete gets fixed

File: pkg/appmgmt/general/general_test.go [hunk 1]
```diff
@@ -614,268 +614,181 @@ func TestGetOwnerReferences(t *testing.T) {
 	assert.Equal(t, returnedOwnerRefs[0].APIVersion, v1.SchemeGroupVersion.String(), "Unexpected owner reference Kind")
 }
 
+type Templete struct {
```
[the remark this answers is not in the document]
Comment (HuangTing-Yao) [reply pkg/appmgmt/general/general_test.go]: Fixed, Thanks.

**Review by craigcondit — APPROVED** [review]
+1 LGTM. Thanks @HuangTing-Yao for the contribution!

**Pull request closed without merging**